bson/src/test/unit/org/bson/BsonBinaryWriterTest.java

```diff
@@ -354,6 +354,56 @@ public void testWriteDBPointer() {
         reader.readEndDocument();
     }
 
+    @Test
+    public void testNullByteInTopLevelName() {
+        writer.writeStartDocument();
+        writer.writeName("a\u0000b");
+        try {
+            writer.writeBoolean(true);
+            fail();
+        } catch (BsonSerializationException e) {
+            // expected
```

Comment:
Let's use `@Test(expected = BsonSerializationException.class)` instead.

Reply:
Resolving, because when `Test.expected` is used, an exception may come from a line different than that being tested. @jyemin pointed this out, and I agree.

Reply:
Talked it over, decide on `assertThrows` instead.

Done.